File: leetcode/golang/154_find_minimum_in_rotated_sorted_array_ii.go
func findMin(nums []int) int {
    l, r := 0, len(nums) - 1
    for l < r {
        i := (l + r) / 2
        if nums[i] > nums[r] {
            l = i + 1
        } else if nums[i] < nums[r] {
            r = i
        } else {
            r--
        }
    }
    return nums[l]
}

File: leetcode/c++/154_find_minimum_in_rotated_sorted_array_ii.cpp
// https://leetcode.cn/problems/find-minimum-in-rotated-sorted-array-ii/description

#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    int findMin(vector<int>& nums) {
        int l = 0, r = nums.size() - 1;
        while (l < r) {
            int i = (l + r) / 2;
            if (nums[i] > nums[r]) l = i +1;
            else if (nums[i] < nums[r]) r = i;
            else r--;
        }
        return nums[l];
    }
};

// 时间复杂度为O(logn)，空间复杂度为O(1)